frontend/src/components/ShootWarnings/ShootWarnings.vue

```diff
@@ -0,0 +1,327 @@
+import { shootItem } from '@/mixins/shootItem'
+import {
+  k8sVersionExpirationForShoot,
+  expiringWorkerGroupsForShoot,
+  isShootHasNoHibernationScheduleWarning,
+  isSelfTerminationWarning
+} from '@/utils'
+import { mapGetters } from 'vuex'
+
+export default {
+  name: 'ShootWarnings',
+  components: {
+    GPopper,
+    K8sExpirationMessage,
+    WorkerGroupExpirationMessage,
+    NoHibernationScheduleMessage,
+    ClusterExpirationMessage,
+    ConstraintWarningMessage
+  },
+  props: {
+    shootItem: {
+      type: Object
+    },
+    small: {
+      type: Boolean,
+      default: false
+    },
+    allWarnings: {
+      type: Boolean,
+      default: false
+    },
+    k8sWarning: {
+      type: Boolean,
+      default: false
+    },
+    machineImageWarning: {
+      type: Boolean,
+      default: false
+    },
+    noHibernationScheduleWarning: {
+      type: Boolean,
+      default: false
+    },
+    clusterExpirationWarning: {
+      type: Boolean,
+      default: false
+    },
+    hibernationConstraintWarning: {
+      type: Boolean,
+      default: false
+    },
+    maintenanceConstraintWarning: {
+      type: Boolean,
+      default: false
+    }
+  },
+  mixins: [shootItem],
+  computed: {
+    ...mapGetters([
+      'canPatchShoots'
+    ]),
+    visible () {
+      return !this.isShootMarkedForDeletion && (
+        this.isK8sWarning ||
+        this.isMachineImageWarning ||
+        this.isNoHibernationScheduleWarning ||
+        this.isClusterExpirationWarning ||
+        this.isConstraintWarning
+      )
+    },
+    warningItems () {
+      const items = []
+      if (this.isK8sWarning) {
+        const { expirationDate, isValidTerminationDate, severity } = this.k8sExpiration
+        items.push({
+          key: 'k8sWarning',
+          icon: 'mdi-update',
+          color: this.colorForSeverity(severity),
+          messageComponent: 'K8sExpirationMessage',
+          messageComponentProperties: {
+            expirationDate,
+            isValidTerminationDate,
+            severity
+          }
+        })
+      }
+      if (this.isMachineImageWarning) {
+        forEach(this.expiredWorkerGroups, ({ expirationDate, isValidTerminationDate, version, name, workerName, severity }) => {
+          items.push({
+            key: `image_${workerName}_${name}`,
+            icon: 'mdi-update',
+            color: this.colorForSeverity(severity),
+            messageComponent: 'WorkerGroupExpirationMessage',
+            messageComponentProperties: {
+              expirationDate,
+              isValidTerminationDate,
+              severity,
+              name,
+              workerName,
+              version
+            }
+          })
+        })
+      }
+      if (this.isNoHibernationScheduleWarning) {
+        items.push({
+          key: 'noHibernationSchedule',
+          icon: 'mdi-calendar-alert',
+          color: 'info',
+          messageComponent: 'NoHibernationScheduleMessage',
```

Comment:
component names should be spcified in the form case as in the html

Reply:
done